File: 0926-flip-string-to-monotone-increasing/0926-flip-string-to-monotone-increasing.cs
public class Solution {
    public int MinFlipsMonoIncr(string s) {
        int flipCount = 0;
        int onesCount = 0;
        
        foreach (var c in s) {
            if (c == '1')
                onesCount++;
            else if (onesCount > flipCount)
                flipCount++;
        }
        return flipCount;
    }
}

File: 0926-flip-string-to-monotone-increasing/0926-flip-string-to-monotone-increasing.py
class Solution:
    def minFlipsMonoIncr(self, s: str) -> int:
        flip_count = number_of_ones_encountered = 0
        for c in s:
            if c == '1':
                number_of_ones_encountered += 1
            elif number_of_ones_encountered > flip_count:
                flip_count += 1

        return flip_count
